vsg/rules/process/rule_029.py

```diff
@@ -15,13 +15,29 @@ def __init__(self):
         self.identifier = '029'
         self.solution = 'Use \'event for clocks.'
         self.phase = 1
+        self.clock = 'event'
+        self.configuration.append('clock')
 
     def _analyze(self, oFile, oLine, iLineNumber):
-        if oLine.insideProcess and re.match('^.*ing_edge\s*\(', oLine.lineLower):
-            self.add_violation(iLineNumber)
+        if self.clock == 'event':
+            if oLine.isClockStatement and re.match('^.*ing_edge\s*\(', oLine.lineLower):
+                self.add_violation(iLineNumber)
+        else:
```

Comment:
yeah, I had wondered that when I wrote that up.  I can change it.

Reply:
Are there are other string configs or are the rest just booleans?  As a user would be nice to see VSG fail right off the bat if it reads the configuration and finds an option that shouldn't be there. Some way of defining a list of valid options for every configuration.   

If that existed, then the else would be fine as we already know inputs are good to go.  Without knowing the work involved that'd be my personal preference.  Can just add that to a TODO somewhere and close this.

Reply:
we have string configs for the upper and lower case and I think it throws an exception if it doesn't match.   I should follow that example.

Reply:
Yep that worked, thanks.  

In the future it might be nice to find some way to bring options into the frontend and check everything at file read time.  Otherwise you could have a lurking bad config and not know about it if your design doesn't happen to contain an issue about that rule.  

But I'm okay approving this pull request with that in mind.